### DISCUSSION TITLE
Custom atom initiated for every unit test

### DISCUSSION THREAD

**Discussion by SomeCodingUser** [pmndrs/jotai#2282] [Q&A]
Hello,

I wrote a custom atom based on the atomWithStorage code which basically does the same but uses an internal api to store the values. 
It all works fine but somehow the atom introduces weird behavior to our unit tests.
When running any unit test which does not depend on this state, the atom is anyway initiated, which throws an error because the api is not reachable.
I tried to mock the response with jest, but the atom is initiated and the code executed before the jest code for mocking is run.

Do you have any idea why an atom would be initiated before test setup is done?

**Comment by dai-shi** [comment]
Does using Provider help?

**Reply by SomeCodingUser** [reply]
Unfortunately not. But I just tried to reproduce it in a new project and it's not reproduceable. So it seems that it's a project related problem and there is some hidden dependency on the atom.

**Reply by SomeCodingUser** [reply]
FYI: we had something like this:
```const myCustomAtom = atomWithCustomStore...

const secondAtom = atom(...)

export const AppState = {
  myCustom: myCustomAtom,
  second: secondAtom,
};
```

One test had a dependency to the second atom which was used like `const second = useAtom(AppState.second)` and this seems to also initiate `myCustomAtom`. Moving `secondAtom` to another file solved the problem.

It seems like adding all atoms to one object is not the best practice, can you confirm that?

**Reply by dai-shi** [reply]
No, I don't think adding all atoms in a single file or in one object is bad.
But maybe in your test scenario, separating files may help.
